### PR TITLE
Fix critic network for Discrete CRR

### LINKED ISSUE #482: Visualize result in real time
Hi, how can you see the results (such as rewards, loss, ...) of the visualization in real-time while training a model? Just like using tensorboard or Visdom. Thanks

### PR DESCRIPTION
**Pull request by nuance1979**
- [x] I have marked all applicable categories:
    + [ ] exception-raising fix
    + [x] algorithm implementation fix
    + [ ] documentation modification
    + [ ] new feature
- [x] I have reformatted the code using `make format` (**required**)
- [x] I have checked the code using `make commit-checks` (**required**)
- [ ] If applicable, I have mentioned the relevant/related issue(s)
- [x] If applicable, I have listed every items in this Pull Request below

This PR:

- Fixes an inconsistency in the implementation of Discrete CRR. Now it uses `Critic` class for its critic, following conventions in other actor-critic policies;
- Updates several offline policies to use `ActorCritic` class for its optimizer to eliminate randomness caused by parameter sharing between actor and critic;
- Adds a `disable_test` option to `offline_trainer` to turn off testing during training;
- Updates the Atari offline results in README.md;
- Moves Atari offline RL examples to `examples/offline`; tests to `test/offline` per review comments.

### CODE REVIEW

File: tianshou/trainer/offline.py [hunk 1]
```diff
@@ -75,12 +77,13 @@ def offline_trainer(
     start_time = time.time()
     test_collector.reset_stat()
 
-    test_result = test_episode(
-        policy, test_collector, test_fn, start_epoch, episode_per_test, logger,
-        gradient_step, reward_metric
-    )
-    best_epoch = start_epoch
-    best_reward, best_reward_std = test_result["rew"], test_result["rew_std"]
+    if test_in_train:
```
Comment (Trinkle23897) [inline tianshou/trainer/offline.py]: I don't think this is the same logic of `test_in_train` compared with other trainers. It is used for testing performance in the middle of an epoch.

**Review by Trinkle23897 — COMMENTED** [review]
Also, could you please move the related result from `examples/atari` to `examples/offline` and move the related test to `test/discrete` to `test/offline`?

File: tianshou/trainer/offline.py (hunk 1 above)
Comment (nuance1979) [reply tianshou/trainer/offline.py]: I changed the name to `disable_test`. I just need a switch to turn off testing in offline setting.

**Comment by nuance1979** [comment]
> Also, could you please move the related result from `examples/atari` to `examples/offline` and move the related test to `test/discrete` to `test/offline`?

Sure. I added a few `__init__.py` files in examples directory so that I can import Atari helpers in `examples/atari` from `examples/offline`.

File: tianshou/trainer/offline.py (hunk 1 above)
Comment (Trinkle23897) [reply tianshou/trainer/offline.py]: So why not use `test_collector=None` to indicate? And this can be also applied to other trainers.

**Comment by Trinkle23897** [comment]
how about the other two trainers?

**Comment by nuance1979** [comment]
> how about the other two trainers?

This functionality is mainly for offline RL when we don't even have an env and want to train a model; other than that, I think it's rarely necessary and/or useful to turn off testing during training.

Of course I can add it if you insist.

**Comment by Trinkle23897** [comment]
Well, actually if the environment simulation is expensive and if we have already know the approximate number of steps to use, excluding the test phase can be more efficient for training an agent.

**Comment by nuance1979** [comment]
> Well, actually if the environment simulation is expensive and if we have already know the approximate number of steps to use, excluding the test phase can be more efficient for training an agent.

Ok. I'll add it then.

**Review by Trinkle23897 — APPROVED** [review]

**Pull request merged**